dapp/src/hooks/useNamesPurchaseMode.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+import { useIotaClient } from '@iota/dapp-kit';
+import { useQuery } from '@tanstack/react-query';
+
+import { useIotaNamesClient } from '@/contexts';
+
+import { queryKey } from './queryKey';
+
+export function useNamesPurchaseMode() {
+    const { iotaNamesClient } = useIotaNamesClient();
+    const iotaClient = useIotaClient();
+
+    const authKeyType = `${iotaNamesClient.config.packageId}::iota_names::AuthKey`;
+
+    const paymentType = `${authKeyType}<${iotaNamesClient.config.paymentsPackageId}::payments::PaymentsAuth>`;
+    const auctionType = `${authKeyType}<${iotaNamesClient.config.auctionPackageId}::auction::AuctionAuth>`;
+
+    return useQuery({
+        // eslint-disable-next-line @tanstack/query/exhaustive-deps
+        queryKey: queryKey.purchaseConfig(paymentType, auctionType),
```

Comment:
For the sake of code consistence, not that it makes any difference at runtime.
```suggestion
        queryKey: [...queryKey.purchaseConfig(paymentType, auctionType)],
```